### DISCUSSION TITLE
How to run dev branch

### DISCUSSION THREAD

**Discussion by osi1880vr** [hlky/stable-diffusion-webui#1025] [Q&A]
I checked out the dev Branch and started ist but it wont start =( 
it is complaining about no finding the frontend module.

Relauncher: Launching...
Traceback (most recent call last):
  File "scripts/webui.py", line 6, in <module>
    from frontend.frontend import draw_gradio_ui
ModuleNotFoundError: No module named 'frontend'
Relauncher: Process is ending. Relaunching in 1s...
        Relaunch count: 22

the files are in place but something prevents it from loading.

Any Idea what I do wrong?

**Comment by codedealer** [comment]
Somewhere else it was suggested that it's a name conflict in conda environments. The environment.yaml on dev has default name "ldm" while installation instructions recommend changing it to "ldo".

I'd suggest keeping `master` and `dev` in separate folders in different conda environments if you plan to use both. Otherwise you should recreate environment after checking out `dev` from scratch instead of updating it via `webui.cmd`.
To recreate conda environment:
1

```
conda env remove -n <your environment name> 
```
2 move your upscaler models out of their respective folders in `src`
3 then delete `src` folder
4
```
conda env create -n <your environment name> -f environment.yaml
```
Move your models back into src, now you should a proper `dev` environment.

**Comment by osi1880vr** [comment]
Hello Thomas, 

 

I will try this out. Thank you very much for your hints.

What did not push me in this direction was the fact that it does not find frontend which is a local file in the first place.

 

But I will try and let you guys know what happend.

To have this multiple envornments is easy to mess up 😃

 

Cheers

Werner

 

Von: Thomas Mello ***@***.***> 
Gesendet: Sonntag, 11. September 2022 19:34
An: sd-webui/stable-diffusion-webui ***@***.***>
Cc: osi1880vr ***@***.***>; Author ***@***.***>
Betreff: Re: [sd-webui/stable-diffusion-webui] How to run dev branch (Discussion #1025)

 

Somewhere else it was suggested that it's a name conflict in conda environments. The environment.yaml on dev has default name "ldm" while installation instructions recommend changing it to "ldo".

I'd suggest keeping master and dev in separate folders in different conda environments if you plan to use both. Otherwise you should recreate environment after checking out dev from scratch instead of updating it via webui.cmd.
To recreate conda environment:
1

conda env remove -n <your environment name> 

2 move your upscaler models out of their respective folders in src
3 then delete src folder
4

conda env create -n <your environment name> -f environment.yaml

Move your models back into src, now you should a proper dev environment.

**Comment by mchaker** [comment]
If you are on linux, you can also try running `export PYTHONPATH="."` before running relauncher.py

**Comment by Michoko92** [comment]
I'm facing the same issue on Windows. Did recreating the environment fix the issue? (I'm on a limited internet connection right now, so I prefer to ask before redownloading everything). Thanks!

**Comment by osi1880vr** [comment]
No it did not help, the issue ist he same.

I still think must be different, it is missing a local file nothing from the env.

 

 

 

Von: Michoko ***@***.***> 
Gesendet: Montag, 12. September 2022 10:37
An: sd-webui/stable-diffusion-webui ***@***.***>
Cc: osi1880vr ***@***.***>; Author ***@***.***>
Betreff: Re: [sd-webui/stable-diffusion-webui] How to run dev branch (Discussion #1025)

 

I'm facing the same issue on Windows. Did recreating the environment fix the issue? (I'm on a limited internet connection right now, so I prefer to ask before redownloading everything). Thanks!

**Comment by Michoko92** [comment]
That's what I suspected, thank you. 🙏

Edit: nice profile pic, by the way 😉

**Comment by osi1880vr** [comment]
It is strange…

 

When I run a scrip in scripts which does this:

 

for d in sys.path:

    print(d)

 

 

then I got for the master version of the webui this output:

 

D:\stableDiffusion\stable-diffusion-webui\scripts

C:\Users\user\miniconda3\envs\ldm\python38.zip

C:\Users\ user \miniconda3\envs\ldm\DLLs

C:\Users\ user \miniconda3\envs\ldm\lib

C:\Users\ user \miniconda3\envs\ldm

C:\Users\ user \miniconda3\envs\ldm\lib\site-packages

d:\stablediffusion\stable-diffusion-webui\src\gfpgan

d:\stablediffusion\stable-diffusion-webui\src\taming-transformers

d:\stablediffusion\stable-diffusion-webui\src\realesrgan

d:\stablediffusion\stable-diffusion-webui

d:\stablediffusion\stable-diffusion-webui\src\k-diffusion

d:\stablediffusion\stable-diffusion-webui\src\clip

 

When I run the same script in the dev environment I get this:

 

D:\stableDiffusion\sd-webui-dev\scripts

C:\Users\ user \miniconda3\envs\ldo-webui-dev\python38.zip

C:\Users\ user \miniconda3\envs\ldo-webui-dev\DLLs

C:\Users\ user \miniconda3\envs\ldo-webui-dev\lib

C:\Users\ user \miniconda3\envs\ldo-webui-dev

C:\Users\ user \miniconda3\envs\ldo-webui-dev\lib\site-packages

 

So my question is why the heck ist he lower part missing in the pathes?

I did nothing different.

Create a folder checkou the webui change the branch and run the webui.cmd

 

Iid this on purpose on an conda env which does not exists.

 

What causes the difference in the number of known path in sys.path?

 

Thanks for any hints.

 

Werner

 

Von: Michoko ***@***.***> 
Gesendet: Montag, 12. September 2022 12:51
An: sd-webui/stable-diffusion-webui ***@***.***>
Cc: osi1880vr ***@***.***>; Author ***@***.***>
Betreff: Re: [sd-webui/stable-diffusion-webui] How to run dev branch (Discussion #1025)

 

That's what I suspected, thank you. 🙏

**Comment by osi1880vr** [comment]
OK funny news so far:

 

I renamed my conda env from ldo-webgui-dev to ldd and build that new env.

 

Now the sys.path of python is much more full

 

D:\stableDiffusion\sd-webui-dev\scripts

C:\Users\user\miniconda3\envs\ldo_webui_dev\python38.zip

C:\Users\ user \miniconda3\envs\ldo_webui_dev\DLLs

C:\Users\ user \miniconda3\envs\ldo_webui_dev\lib

C:\Users\ user \miniconda3\envs\ldo_webui_dev

C:\Users\ user \miniconda3\envs\ldo_webui_dev\lib\site-packages

d:\stablediffusion\sd-webui-dev\src\gfpgan

d:\stablediffusion\sd-webui-dev\src\taming-transformers

d:\stablediffusion\sd-webui-dev\src\realesrgan

d:\stablediffusion\sd-webui-dev\src\latent-diffusion

d:\stablediffusion\sd-webui-dev\src\k-diffusion

d:\stablediffusion\sd-webui-dev\src\clip

 

the only one missing from the master versions path is:

 

d:\stablediffusion\sd-webui-dev

 

Now I do something nasty.

 

In webui.py I add this line as the very first line oft he file

 

sys.path.insert(0, os.getcwd())

 

 

Now I got the dev version running.

This is not how it is meant tob e, there ist still something wrong which Im causing but I dont see it yet.

 

But ist working and so I will have a look into the dev version fort he first time 😃

 

 

 

 

Von: Michoko ***@***.***> 
Gesendet: Montag, 12. September 2022 12:51
An: sd-webui/stable-diffusion-webui ***@***.***>
Cc: osi1880vr ***@***.***>; Author ***@***.***>
Betreff: Re: [sd-webui/stable-diffusion-webui] How to run dev branch (Discussion #1025)

 

That's what I suspected, thank you. 🙏

**Comment by Michoko92** [comment]
Wow, adding :

`sys.path.insert(0, os.getcwd())`

after the first import in webui.py did fix the issue for me too! Good job! 🥇

**Comment by osi1880vr** [comment]
But ist only a patch to the symptome where I fail to find the cause.

But for now it works 😊

 

Von: Michoko ***@***.***> 
Gesendet: Montag, 12. September 2022 16:28
An: sd-webui/stable-diffusion-webui ***@***.***>
Cc: osi1880vr ***@***.***>; Author ***@***.***>
Betreff: Re: [sd-webui/stable-diffusion-webui] How to run dev branch (Discussion #1025)

 

Wow, adding :

sys.path.insert(0, os.getcwd())

after the first import in webui.py did fix the issue for me too! Good job! 🥇
